realsense_ros_camera/src/base_realsense_node.cpp

```diff
@@ -917,32 +925,179 @@ void BaseRealSenseNode::publish_static_tf(const ros::Time& t,
     msg.header.stamp = t;
     msg.header.frame_id = from;
     msg.child_frame_id = to;
-    msg.transform.translation.x = trans.z;
-    msg.transform.translation.y = -trans.x;
-    msg.transform.translation.z = -trans.y;
+    msg.transform.translation.x = -trans.z;
+    msg.transform.translation.y = trans.x;
+    msg.transform.translation.z = trans.y;
     msg.transform.rotation.x = q.x;
     msg.transform.rotation.y = q.y;
     msg.transform.rotation.z = q.z;
     msg.transform.rotation.w = q.w;
     _static_tf_broadcaster.sendTransform(msg);
 }
 
+void BaseRealSenseNode::prepareTransforms()
+{
+    // Publish transforms for the cameras
+    ros::Rate loop_rate(_tf_publication_rate);
+
+    while (ros::ok())
+    {
+      publishDynamicTransforms();
+
+      loop_rate.sleep();
+    }
+}
+
+void BaseRealSenseNode::publishDynamicTransforms()
+{
+    // For single camera, we can use publishStaticTransforms() to publish static transform.
+    // However, for multiple cameras we must use publishDynamicTransforms().
+
+    // In our URDF, we provide the tf from camera_link to camera_depth_frame. The driver
+    // considers depth frame as reference frame and provides other tf publication,
+    // which are tf from depth to color, color to color_optical,  depth to depth_optical,
+    // depth to ir and ir to ir_optical
+
+    ros::Time transform_ts_ = ros::Time::now();
+    tf::Transform tr_;
+    tf::Transform tr_optical_;
+
+    tf::Quaternion q_optical_;
+    q_optical_.setRPY(-M_PI / 2, 0.0, -M_PI / 2);
+    tr_optical_.setOrigin(tf::Vector3(0, 0, 0));
+    tr_optical_.setRotation(q_optical_);
+
+    _dynamic_tf_broadcaster.sendTransform(tf::StampedTransform(tr_optical_, transform_ts_,
+        _frame_id[DEPTH], _optical_frame_id[DEPTH]));
+
+    if (_enable[COLOR])
+    {
+        // Transform depth to color
+        auto& ex = (_align_depth)?(_i_ex):(getRsExtrinsics(DEPTH, COLOR));
+        auto Q = rotationMatrixToQuaternion(ex.rotation);
+
+        tr_.setOrigin(tf::Vector3(-ex.translation[2], ex.translation[0], ex.translation[1]));
+        tf::Quaternion q(Q.x(), Q.y(), Q.z(), Q.w());
+        tr_.setRotation(q);
+
+        _dynamic_tf_broadcaster.sendTransform(tf::StampedTransform(tr_, transform_ts_,
+            _frame_id[DEPTH], _frame_id[COLOR]));
+
+        // Transform color frame to color optical frame
+        _dynamic_tf_broadcaster.sendTransform(tf::StampedTransform(tr_optical_, transform_ts_,
+            _frame_id[COLOR], _optical_frame_id[COLOR]));
+
+        if (_align_depth)
+        {
+            _dynamic_tf_broadcaster.sendTransform(tf::StampedTransform(tr_, transform_ts_,
```

Comment:
is the _ name and /name _ our coding standard??? seems a bit confusing... i'm used to local variables being without any _

Reply:
I followed the convention of the driver. They use _var for global variables and var_ for some of the local variables. I agree with remove all the underscore from local ones